src/routes/safe/components/Apps/hooks/useAppList.ts

```diff
@@ -12,17 +12,66 @@ type onAppRemovedHandler = (appId: string) => void
 
 type UseAppListReturnType = {
   appList: SafeApp[]
-  loadingAppList: boolean
   onAppToggle: onAppToggleHandler
   onAppAdded: onAppAddedHandler
   onAppRemoved: onAppRemovedHandler
 }
 
 const useAppList = (): UseAppListReturnType => {
   const [appList, setAppList] = useState<SafeApp[]>([])
-  const [loadingAppList, setLoadingAppList] = useState<boolean>(true)
+
+  const onAppToggle: onAppToggleHandler = useCallback(
+    async (appId, enabled) => {
+      // update in-memory list
+      const appListCopy = [...appList]
+
+      const app = appListCopy.find((a) => a.id === appId)
+      if (!app) {
+        return
+      }
+      app.disabled = !enabled
+
+      setAppList(appListCopy)
+
+      // update storage list
+      const listToPersist: StoredSafeApp[] = appListCopy.map(({ url, disabled }) => ({ url, disabled }))
+      saveToStorage(APPS_STORAGE_KEY, listToPersist)
+    },
+    [appList],
+  )
+
+  const onAppAdded: onAppAddedHandler = useCallback(
+    (app) => {
+      const newAppList = [
+        { url: app.url, disabled: false },
+        ...appList.map((a) => ({
+          url: a.url,
+          disabled: a.disabled,
+        })),
+      ]
+      saveToStorage(APPS_STORAGE_KEY, newAppList)
+
+      setAppList([...appList, { ...app, isDeletable: true }])
+    },
+    [appList],
+  )
+
+  const onAppRemoved: onAppRemovedHandler = useCallback(
+    (appId) => {
+      const appListCopy = appList.filter((a) => a.id !== appId)
+
+      setAppList(appListCopy)
+
+      const listToPersist: StoredSafeApp[] = appListCopy.map(({ url, disabled }) => ({ url, disabled }))
+      saveToStorage(APPS_STORAGE_KEY, listToPersist)
+    },
+    [appList],
```

Comment:
I may be misunderstanding the use of `useCallback`, so please be gentle :smile:, but if on every function call you modify the dependency element, what's the point of memoizing it after all?

`onAppToggle`, `onAppAdded` and `onAppRemoved`, the three do the same.